src/plugins/intel_cpu/tests/functional/shared_tests_instances/single_layer_tests/is_inf.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <vector>
+#include "single_layer_tests/is_inf.hpp"
+
+using namespace ov::test;
+using namespace ov::test::subgraph;
+
+namespace {
+std::vector<std::vector<InputShape>> inShapesStatic = {
+        { {{}, {{2}}} },
+        { {{}, {{2, 200}}} },
+        { {{}, {{10, 200}}} },
+        { {{}, {{1, 10, 100}}} },
+        { {{}, {{4, 4, 16}}} },
+        { {{}, {{1, 1, 1, 3}}} },
+        { {{}, {{2, 17, 5, 4}}} },
+        { {{}, {{2, 17, 5, 1}}} },
+        { {{}, {{1, 2, 4}}} },
+        { {{}, {{1, 4, 4}}} },
+        { {{}, {{1, 4, 4, 1}}} },
+        { {{}, {{16, 16, 16, 16, 16}}} },
+        { {{}, {{16, 16, 16, 16, 1}}} },
+        { {{}, {{16, 16, 16, 1, 16}}} },
+        { {{}, {{16, 32, 1, 1, 1}}} },
+        { {{}, {{1, 1, 1, 1, 1, 1, 3}}} },
+        { {{}, {{1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}}} }
+};
+
+std::vector<std::vector<InputShape>> inShapesDynamic = {
+        {{{ngraph::Dimension(1, 10), 200}, {{2, 200}, {1, 200}}}}
+};
+
+std::vector<ElementType> netPrecisions = {
+        ov::element::f32
+};
+
+std::vector<bool> detectNegative = {
+    true, false
+};
+
+std::vector<bool> detectPositive = {
+    true, false
+};
+
+std::map<std::string, std::string> additional_config = {};
+
+const auto isInfParams = ::testing::Combine(
+        ::testing::ValuesIn(inShapesStatic),
+        ::testing::ValuesIn(detectNegative),
+        ::testing::ValuesIn(detectPositive),
+        ::testing::ValuesIn(netPrecisions),
+        ::testing::Values(CommonTestUtils::DEVICE_CPU),
+        ::testing::Values(additional_config));
+
+const auto isInfParamsDyn = ::testing::Combine(
+        ::testing::ValuesIn(inShapesDynamic),
+        ::testing::ValuesIn(detectNegative),
+        ::testing::ValuesIn(detectPositive),
+        ::testing::ValuesIn(netPrecisions),
+        ::testing::Values(CommonTestUtils::DEVICE_CPU),
+        ::testing::Values(additional_config));
+
+
+TEST_P(IsInfLayerTest, CompareWithRefs) {
```

Comment:
Hello @nshchego, I'm working on adding support for these three operations (IsFinite, IsInf, IsNaN) to the GPU plugin, and I plan to use the "ComparisonLayerTest" class as well as this new "IsInfLayerTest" class. Is this intended to create separate "TEST_P(IsInfLayerTest, CompareWithRefs)" for CPU, and should I also create one for GPU tests? Or it can be moved to "src/tests/functional/plugin/shared/include/single_layer_tests/comparison.hpp" or to maybe its own new "src/tests/functional/plugin/shared/include/single_layer_tests/is_inf.hpp" so we can reuse the same test in all plugins?